app/controllers/Application.java

```diff
@@ -1137,14 +1138,26 @@ public static Result restFlowGraphData(String flowDefId) {
 
       // Execution record
       JsonObject dataset = new JsonObject();
-      dataset.addProperty("flowtime", mrJobsList.get(mrJobsList.size() - 1).finishTime);
+
+      // TODO should make it specific to langoustine as other scheduler are not storing the date in flowExecId field
+      // Parsing the langoustine date
+      long maybeParsed;
+      try {
+        maybeParsed = sdf.parse(mrJobsList.get(mrJobsList.size() - 1).flowExecId).getTime();
+      } catch (ParseException e) {
+        logger.info("Could not parse " + mrJobsList.get(mrJobsList.size() - 1).flowExecId);
+        maybeParsed = mrJobsList.get(mrJobsList.size() - 1).finishTime;
+      }
+      dataset.addProperty("flowtime", maybeParsed);
       dataset.addProperty("score", flowPerfScore);
       dataset.add("jobscores", jobScores);
 
       datasets.add(dataset);
     }
 
-    return ok(new Gson().toJson(datasets));
+    JsonArray sortedDatasets = sortJsonArray(datasets);
```

Comment:
You are not breaking any existing behaviour with other scheduler by doing this way?